geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/AggregatingScan.scala

```diff
@@ -78,14 +81,18 @@ trait AggregatingScan[T <: AnyRef { def isEmpty: Boolean; def clear(): Unit }]
     // noinspection LanguageFeature
     result.clear()
     while (hasNextData && notFull(result)) {
-      nextData(setValues)
-      if (validateFeature(reusableSf)) {
-        // write the record to our aggregated results
-        if (hasTransform) {
-          aggregateResult(reusableTransformSf, result)
-        } else {
-          aggregateResult(reusableSf, result)
+      try {
```

Comment:
Do we believe this would cause any performance impact?  (I think it is the right thing to do, etc. just asking...)

Reply:
trys that don't throw anything are generally considered to be "free"...